proxy/src/main.rs

```diff
@@ -36,11 +36,26 @@ pub async fn main() -> anyhow::Result<()> {
     banner::print_banner();
 
     let config = Config::load()?;
-    let client = http_client::build(
+    let retry_policy = reqwest::retry::for_host(config.broker_uri.host_str().unwrap().to_string())
+        .classify_fn(|res|  {
+            if res.method() != reqwest::Method::GET {
+                return res.success();
+            }
+            if let Some(StatusCode::BAD_GATEWAY | StatusCode::SERVICE_UNAVAILABLE | StatusCode::GATEWAY_TIMEOUT) = res.status() {
+                res.retryable()
+            } else if res.error().and_then(|e| e.downcast_ref::<reqwest::Error>()).is_some_and(reqwest::Error::is_timeout) {
+                res.retryable()
+            } else {
+                res.success()
+            }
+        })
+        .max_retries_per_request(3);
```

Comment:
It would be nice if we could add more details to the retry logic, e.g., back-off timings, jitter, etc. I'm a bit concerned that multiple requests fail and retry at the same time, hence putting even more pressure on the reverse proxy. Unfortunately, I don't see a straightforward way in Reqwest.